src/main/java/programminglife/model/Graph.java

```diff
@@ -0,0 +1,127 @@
+package programminglife.model;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.*;
+
+/**
+ * Created by marti_000 on 25-4-2017.
+ */
+public class Graph {
+    private String id;
+    private Set<Node> rootNodes;
+    private List<Node> nodes;
```

Comment:
This needs javadoc specifying the invariant (the nodes are placed on the place of their id)